flow/surface_frame.h

```diff
@@ -21,17 +21,52 @@ class SurfaceFrame {
   using SubmitCallback =
       std::function<bool(const SurfaceFrame& surface_frame, SkCanvas* canvas)>;
 
+  // Information about the underlying framebuffer
+  struct FramebufferInfo {
+    // Indicates whether or not the surface supports pixel readback as used in
+    // circumstances such as a BackdropFilter.
+    bool supports_readback = false;
+
+    // This is the area of framebuffer that lags behind the front buffer.
+    //
+    // Correctly providing exiting_damage is necessary for supporting double and
+    // triple buffering. Embedder is responsible for tracking this area for each
+    // of the back buffers used. When doing partial redraw, this area will be
+    // repainted alongside of dirty area determined by diffing current and
+    // last successfully rasterized layer tree;
+    //
+    // If existing damage is unspecified (empty vector), entire frame will be
+    // rasterized (no partial redraw). To signal that there is no existing
+    // damage use vector with an empty SkIRect.
+    std::vector<SkIRect> existing_damage;
+  };
+
   SurfaceFrame(sk_sp<SkSurface> surface,
-               bool supports_readback,
+               FramebufferInfo framebuffer_info,
```

Comment:
Consider making this a `std::unique_ptr<FrameBufferInfo>` given that we will be passing around a `vector`.

Reply:
Is it necessary even if we go with single rect? Also even with vector, wouldn't `std::move` be enough in places where we know the vector is not empty? I don't think that extra heap allocation is worth it if we can just move the `framebuffer_info` and thus the vector itself.

Reply:
Sounds good.